src/backend/vulkan/src/physical_device.rs

```diff
@@ -12,162 +12,673 @@ use hal::{
     queue, DescriptorLimits, DynamicStates, Features, Limits, PhysicalDeviceProperties,
 };
 
-use std::{ffi::CStr, fmt, mem, sync::Arc, unreachable};
+use std::{ffi::CStr, fmt, mem, ptr, sync::Arc, unreachable};
 
 use crate::{
     conv, info, Backend, Device, DeviceExtensionFunctions, Queue, QueueFamily, RawDevice,
     RawInstance, Version,
 };
 
-pub(crate) fn map_device_features(
```

Comment:
Note: moved to `PhysicalDeviceFeatures::from_extensions_and_requested_features` and changed to also look at the api version and enabled extensions list